eth/vm/forks/london/computation.py

```diff
@@ -11,3 +18,62 @@ class LondonComputation(BerlinComputation):
     Inherits from :class:`~eth.vm.forks.berlin.BerlinComputation`
     """
     opcodes = LONDON_OPCODES
+
+    @classmethod
+    def apply_create_message(
+        cls,
+        state: StateAPI,
+        message: MessageAPI,
+        transaction_context: TransactionContextAPI
+    ) -> ComputationAPI:
+
+        snapshot = state.snapshot()
+
+        # EIP161 nonce incrementation
+        state.increment_nonce(message.storage_address)
+
+        computation = cls.apply_message(state, message, transaction_context)
+
+        if computation.is_error:
+            state.revert(snapshot)
+            return computation
+        else:
+            contract_code = computation.output
+
+            if contract_code and len(contract_code) >= EIP170_CODE_SIZE_LIMIT:
+                computation.error = OutOfGas(
+                    f"Contract code size exceeds EIP170 limit of {EIP170_CODE_SIZE_LIMIT}."
+                    f" Got code of size: {len(contract_code)}"
+                )
+                state.revert(snapshot)
+            elif contract_code:
+                contract_code_gas_cost = len(contract_code) * constants.GAS_CODEDEPOSIT
+                try:
+                    computation.consume_gas(
+                        contract_code_gas_cost,
+                        reason="Write contract code for CREATE / CREATE2",
+                    )
+                except OutOfGas as err:
+                    computation.error = err
+                    state.revert(snapshot)
+                else:
+                    if contract_code[:1] == EIP3541_RESERVED_STARTING_BYTE:
+                        # As per EIP-3541, gas is still consumed on a revert of this nature
+                        state.revert(snapshot)
+                        raise ReservedBytesInCode(
+                            "Contract code begins with EIP3541 reserved byte '0xEF'."
+                        )
+                    else:
```

Comment:
Looks like this is the full change? It's a lot of work to comb through to confirm that. I would love to see that get easier. Here's another option (I don't love it, but it gets at the idea of isolating the change in the new London code):
1. in the Spurious Dragon computation, add a `def validate_new_code(cls, contract_code: bytes): pass`
2. Run `cls.validate_new_code(contract_code)` just before the `SETTING CODE` log. (validate_X in py-evm is generally understood to mean: raise on failure)
2. Add a `def validate_new_code` in London that checks the `0xEF` byte.

I'm imagining the validate call to look like:
```py
except OutOfGas as err:
  ...
else:
  try:
    cls.validate_new_code(contract_code)
  except VMError:
    state.revert(snapshot)
    raise

  if cls.logger:
    cls.logger.debug2("SETTING CODE: " ...
```
(The thing I don't like about it is that it continues to embrace the inheritance style, but it's also pervasive in py-evm already. So it may be our best option)

Reply:
@carver That was indeed the full change. This is definitely better. My goal was  to refactor this at some point but thought I'd track it as a separate issue and just get these changes in. This seems quick enough to implement in this PR though and will make it all much easier to read too 👍

Reply:
Great :)